VertexCompositeProducer/src/LamC3PFitter.cc

```diff
@@ -417,43 +471,49 @@ void LamC3PFitter::fitLamCCandidates(
       if( !trkTSCP1.isValid() || !trkTSCP2.isValid() ) continue;
 
       double totalE1 = sqrt( trkTSCP1.momentum().mag2() + protonMassLamC3PSquared ) +
-                      sqrt( trkTSCP2.momentum().mag2() + kaonMassLamC3PSquared );
+                      sqrt( trkTSCP2.momentum().mag2() + piMassLamC3PSquared );
       double totalE1Sq = totalE1*totalE1;
 
-      double totalE2 = sqrt( trkTSCP1.momentum().mag2() + kaonMassLamC3PSquared ) +
+      double totalE2 = sqrt( trkTSCP1.momentum().mag2() + piMassLamC3PSquared ) +
                       sqrt( trkTSCP2.momentum().mag2() + protonMassLamC3PSquared );
       double totalE2Sq = totalE2*totalE2;
 
       double totalPSq =
         ( trkTSCP1.momentum() + trkTSCP2.momentum() ).mag2();
 
-//      double totalPt =
-//        ( trkTSCP1.momentum() + trkTSCP2.momentum() ).perp();
-
       double mass1 = sqrt( totalE1Sq - totalPSq);
       double mass2 = sqrt( totalE2Sq - totalPSq);
 
-      if( (mass1 > mKPCutMax || mass1 < mKPCutMin) && (mass2 > mKPCutMax || mass2 < mKPCutMin)) continue;
-//      if( totalPt < dPtCut ) continue;
+      if( (mass1 > mPiPCutMax || mass1 < mPiPCutMin) && (mass2 > mPiPCutMax || mass2 < mPiPCutMin)) continue;
+*/
 
       for(unsigned int trdx3 = 0; trdx3 < theTrackRefs_sgn2.size(); trdx3++) {
 
+        std::map<std::string, float> theMTDtrackInfo3 = theMTDtrackInfo_sgn2[trdx3]; // mtd
+        bool isKaon3 = theMTDtrackInfo3["isKaon"];
+        if(!isKaon3) continue;
```

Comment:
You can combine these 3 lines as:
if (theMTDtrackInfo_sgn2[trdx3]["isKaon"]!=0.) coninue;